spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/KubernetesBootstrapConfiguration.java

```diff
@@ -16,16 +16,69 @@
 
 package org.springframework.cloud.kubernetes.commons.config;
 
+import org.aspectj.lang.annotation.Aspect;
+
+import org.springframework.boot.autoconfigure.aop.AopAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.cloud.kubernetes.commons.ConditionalOnKubernetesEnabled;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.retry.annotation.EnableRetry;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.retry.interceptor.RetryInterceptorBuilder;
+import org.springframework.retry.interceptor.RetryOperationsInterceptor;
+import org.springframework.retry.policy.NeverRetryPolicy;
 
 /**
  * @author Ryan Baxter
+ * @author Isik Erhan
  */
 @Configuration(proxyBeanMethods = false)
 @ConditionalOnKubernetesEnabled
 @EnableConfigurationProperties({ ConfigMapConfigProperties.class, SecretsConfigProperties.class })
 public class KubernetesBootstrapConfiguration {
 
+	@ConditionalOnKubernetesConfigPropertiesRetryEnabled
+	@ConditionalOnClass({ Retryable.class, Aspect.class, AopAutoConfiguration.class })
+	@Configuration(proxyBeanMethods = false)
+	@EnableRetry(proxyTargetClass = true)
+	@Import(AopAutoConfiguration.class)
+	static class RetryConfiguration {
+
+		private static RetryOperationsInterceptor retryOperationsInterceptor(
+				AbstractConfigProperties.RetryProperties retryProperties) {
+			return RetryInterceptorBuilder.stateless().backOffOptions(retryProperties.getInitialInterval(),
+					retryProperties.getMultiplier(), retryProperties.getMaxInterval())
+					.maxAttempts(retryProperties.getMaxAttempts()).build();
+		}
+
+		@Bean
+		@ConditionalOnConfigMapPropertiesRetryEnabled
+		public RetryOperationsInterceptor configMapPropertiesRetryInterceptor(
+				ConfigMapConfigProperties configProperties) {
+			return retryOperationsInterceptor(configProperties.getRetry());
+		}
+
+		@Bean("configMapPropertiesRetryInterceptor")
```

Comment:
can I ask why you name these with the same name, though one does not retry and the other does? Or this is again for consistency across projects?